compiler/src/iree/compiler/GlobalOptimization/DemoteContractionInputsToBF16.cpp

```diff
@@ -24,9 +24,18 @@ namespace {
 // For narrowable inputs, selects
 struct DemoteContractionInputsToBF16Pattern
     : public OpInterfaceRewritePattern<linalg::LinalgOp> {
-  using OpInterfaceRewritePattern::OpInterfaceRewritePattern;
+  using OpInterfaceRewritePattern<linalg::LinalgOp>::OpInterfaceRewritePattern;
+  explicit DemoteContractionInputsToBF16Pattern(
+      MLIRContext *ctx,
+      mlir::iree_compiler::GlobalOptimization::demoteOnly &option)
```

Comment:
style nit: drop `mlir::iree_compiler::GlobalOptimization::`, it is already within the namespace. Same for other places